### DISCUSSION TITLE
message content returns empty results

### DISCUSSION THREAD

**Discussion by wael-sudo2** [Rapptz/discord.py#9461] [Q&A] [answered]
```
async for message in server.history(limit=100):
                messages.append(message.content)
print(messages)

`intents = discord.Intents.default()
intents.guild_messages= True
client = MyClient(intents=intents)
client.run('token')`
```
result -> ['', '', '', '', '', '', '', '', '', '', '', '', '', ''] 14 empty cells instead of their content 
and when i add `intents.message_content = True`
it throws PrivilegedIntentsRequired error altough the bot has all the permissions 
any idea how to solve this ?

**Comment by Rapptz** [comment]
You need to enable the Message Content privileged intent in the [portal for your application](https://discord.com/developers/applications) as mentioned in the error message.

**Reply by wael-sudo2** [reply]
![image](https://github.com/Rapptz/discord.py/assets/106942086/fd4fc55a-6516-4512-b6e5-3994618731c1)
all permissions are enabled

**Reply by tailoric** [reply, marked as answer]
You need to enable the **intents** not permissions 
![Screenshot_20230622_154255_Firefox](https://github.com/Rapptz/discord.py/assets/16943959/43c25184-a394-4e0f-9824-29036191875c)
